src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] hashMapTable;
+    private int size;
+
+    public MyHashMap() {
+        hashMapTable = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
+
+    private class Node<K,V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            result = 31 * result + (key == null ? 0 : key.hashCode());
+            return result;
+        }
+
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> putNode = new Node<>(computeHash(key), key, value, null);
+        Node<K, V> currentNode = hashMapTable[putNode.hash];
+        do {
+            if (currentNode == null) {
+                hashMapTable[putNode.hash] = putNode;
+                size++;
+                extendCapacityIfNecessary();
```

Comment:
You may call you extend method on line 39. Make this check every time you enter put()

Reply:
My logic has been made on check up before as we put an element any way.